integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java

```diff
@@ -139,4 +225,507 @@ public static void deleteDomainResource(String domainNS, String domainUid) {
+          getLogger().severe(podLog);
+          fail("Domain create job failed");
+        }
+      }
+    }
+  }
+
+  /**
+   * Create a WebLogic domain in image using WDT.
+   *
+   * @param domainUid domain uid
+   * @param domainNamespace namespace in which the domain to be created
+   * @param wdtModelFileForDomainInImage WDT model file used to create domain image
+   * @param appSrcDirList list of the app src in WDT model file
+   * @param wlSecretName wls admin secret name
+   * @param clusterName cluster name
+   * @param replicaCount replica count of the cluster
+   * @return oracle.weblogic.domain.Domain object
+   */
+  public static Domain createAndVerifyDomainInImageUsingWdt(String domainUid,
+                                                            String domainNamespace,
+                                                            String wdtModelFileForDomainInImage,
+                                                            List<String> appSrcDirList,
+                                                            String wlSecretName,
+                                                            String clusterName,
+                                                            int replicaCount) {
+
+    // create secret for admin credentials
+    getLogger().info("Create secret for admin credentials");
+    createSecretWithUsernamePassword(wlSecretName, domainNamespace, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // create image with model files
+    getLogger().info("Creating image with model file and verify");
+    String domainInImageWithWDTImage = createImageAndVerify("domaininimage-wdtimage",
+        Collections.singletonList(MODEL_DIR + "/" + wdtModelFileForDomainInImage), appSrcDirList,
+        Collections.singletonList(MODEL_DIR + "/" + WDT_BASIC_MODEL_PROPERTIES_FILE),
+        WEBLOGIC_IMAGE_NAME, WEBLOGIC_IMAGE_TAG, WLS_DOMAIN_TYPE, false,
+        domainUid, false);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(domainInImageWithWDTImage);
+
+    // Create the repo secret to pull the image
+    // this secret is used only for non-kind cluster
+    createOcirRepoSecret(domainNamespace);
+
+    return createDomainInImageAndVerify(domainUid, domainNamespace, domainInImageWithWDTImage, wlSecretName,
+        clusterName, replicaCount);
+  }
+
+  /**
+   * Create domain with domain-in-image type and verify the domain is created.
+   * Also verify the admin server pod and managed server pods are up and running.
+   * @param domainUid domain uid
+   * @param domainNamespace domain namespace
+   * @param imageName image name used to create domain-in-image domain
+   * @param wlSecretName wls admin secret name
+   * @param clusterName cluster name
+   * @param replicaCount replica count of the cluster
+   * @return oracle.weblogic.domain.Domain object
+   */
+  public static Domain createDomainInImageAndVerify(String domainUid,
```

Comment:
make similar changes here - change method name to createDomainResourceForDomainInImage and move createDomainAndVerify outside

Reply:
fixed